### PR TITLE
Issue 3136: Merge fixes from master to r0.4 for release v0.4.0 (Documentation) 

### LINKED ISSUE #3133: Inclusion of content for Retention Policy in Pravega concepts
**Problem description**
1. Need to include content for Retention Policy

**Problem location**
Documentation/Pravega-concepts.md

### PR DESCRIPTION
**Pull request by Aparnarr**
**Change log description**  
Merging #3134, #3102 , #3098, #3097, #3157 into 0.4 release branch for 0.4.0.

**Purpose of the change**  
Closes #3133, #3096, #3095, #3086, #3156, #3142.

### CODE REVIEW

**Comment by Aparnarr** [comment]
transaction
request
controller

**Comment by tkaitchuck** [comment]
This is triggering a Rat failure. (something is missing a licence header or maybe the config file needs to be updated) Either way we can't merge until that is fixed.

File: documentation/src/docs/security/pravega-security-configurations.md [hunk 1]
```diff
@@ -1,54 +1,970 @@
...
-|Parameter|Details|Default Value|
-|---------|-------|-------------|
-|singlenode.enableTls|Enable TLS between all the components deployed within the singlenode| False|
-|singlenode.enableAuth|Enable authentication/authorization between all the components within the singlenode |False|
-|singlenode.certFile|If TLS is enabled, the public key certificate is used for internal communication between segment store and controller|Empty|
-|singlenode.keyFile|If TLS is enabled, this represents the private key by all the server sockets| Empty|
-|singlenode.passwdFile|If _auth_ is enabled, this represents the password file for the default _auth_ plugin implementation|Empty|
-|singlenode.userName|If _auth_ is enabled, this represents the username used for internal communication between segment store and controller|Empty|
-|singlenode.passwd|If _auth_ is enabled, this represents the password used for internal communication between segment store and controller|Empty|
+
```
Comment (shrids) [inline documentation/src/docs/security/pravega-security-configurations.md]: The missing copyright header caused the failure of `:bindings:rat` gradle task. The contents of the file needs to be relooked into.

**Comment by Aparnarr** [comment]
@tkaitchuck  Please check!

File: documentation/src/docs/security/pravega-security-configurations.md (hunk 1 above)
Comment (Aparnarr) [reply documentation/src/docs/security/pravega-security-configurations.md]: @shrids @tkaitchuck Thanks Sandeep and Tom.... The file `pravega-security-configurations.md` was corrupted..have fixed it..Please check for merging

**Pull request merged**